src/OpenSearch/Handlers/SigV4Handler.php

```diff
@@ -103,13 +103,17 @@ private function createRingRequest(RequestInterface $request, array $originalReq
             $body = null;
         }
 
+        // Reset the explicit port in the URL
+        $client = $originalRequest['client'];
+        unset($client['curl'][CURLOPT_PORT]);
```

Comment:
Is this modifying the value inside `$originalRequest['client']['curl']` that is passed in, aka has a side effect of modifying a caller value? Shouldn't this structure be DUP-ed and the value be overwritten with the port being explicitly set instead?

Reply:
Yes the problem is when the URL does not contain a port we force to use 9200 see here
https://github.com/opensearch-project/opensearch-php/blob/main/src/OpenSearch/Connections/Connection.php#L141

I would like to get rid of that behavior in the next major version of the sdk and use default URL ports https without port = 443

As the Sigv4Handler is only used against AWS Endpoint which was in my case always a https without port, I would like to drop it here first and then change that logic in future :)

Reply:
KK, care to open a bug for this? I think modifying input is my biggest problem, I don't expect side effects within a function like this

Reply:
I opend a issue https://github.com/opensearch-project/opensearch-php/issues/86